### PR TITLE
ci: add script to generate dark reader ignore configuration

### PR DESCRIPTION
**Pull request by Magniquick**
This creates a [dark-reader.ts](https://github.com/catppuccin/userstyles/compare/main...Magniquick:userstyles:main?expand=1#diff-2fc254f52acd99fd374dbc287531c3b205d937208dad5a67585d8a554ec623ac) that traverses the entire repository to create exclusion rules for dark reader.
Initially written in python, rewritten in deno following popular demand.
This still breaks on anything more complex than a bare domain, (probably) due to escaping shenanigans with regex.
eg: google.com is broken, but mail.google.com works.

Basically #962, without the stupid pull bot messing up things. 
Ill squash the commits later.

### CODE REVIEW

**Comment by uncenter** [comment]
Closing this attempt for now, but it is on the Roadmap and we will revisit this in the future when I have more time!
